Game.java

```diff
@@ -35,87 +29,89 @@ private List<String> getAllQuestionFileNames() {
-                System.out.print((i+1) + " ");
-                System.out.println(quiz.getChoices().get(i));
-            }
+            Quiz quiz = QuizLoader.load(dirname + "/" + getQuizFileName());
             
-            long startTime = System.currentTimeMillis();
-            while (true) {
-                if (System.currentTimeMillis() - startTime > 10000) {
-                    System.out.println("時間切れ");
-                    System.out.println("正解は" + quiz.getAnswer());
-                    break;
-                }
-                if (isr.ready()) {
-                    String choice = br.readLine();
-                    if (quiz.check(choice)) {
-                        System.out.println("正解");
-                        ++solved;
-                    } else {
-                        System.out.println("不正解");
-                        System.out.println("正解は" + quiz.getAnswer());
-                    }
-                    break;
-                }
-                Thread.sleep(100);
-            }
+            ++challenged;
+            dumpQuiz(quiz);
+            if(checkAnswer(isr, br, quiz)) ++solved;
 
             System.out.println("現在の正解数 " + solved + "問(" + challenged +"問中)"); 
             if (fileNames.size() == challenged) {
                 System.out.println("終わり！");
                 break;
             }
-            System.out.println("つづけますか (y/n)");
-            boolean exitFlag = false;
-            while(true){
-                String line = br.readLine();
-                if(line.equals("y")) break;
-                if(line.equals("n")){
-                    exitFlag = true;
-                    break;
+            
+            if(!checkExit(br)) break;
+        }
+        System.out.println("またあそんでねー");
+    }
+
+    private boolean checkAnswer(InputStreamReader isr, BufferedReader br, Quiz quiz)
+            throws IOException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        System.out.print("答えを書いてね! : ");
+        while (true) {
+            if (System.currentTimeMillis() - startTime > 10000) {
+                System.out.println("時間切れ");
+                System.out.println("正解は" + quiz.getAnswer());
+                break;
+            }
+
+            if (isr.ready()) {
+                String choice = br.readLine();
+                if (quiz.check(choice)) {
+                    System.out.println("正解");
+                    return true;
+                } else {
+                    System.out.println("不正解");
+                    System.out.println("正解は" + quiz.getAnswer());
+                    return false;
                 }
-                System.out.println("y もしくは nを入力してください");
             }
-            if(exitFlag) break;
+            Thread.sleep(100);
         }
-        System.out.println("またあそんでねー");
-        // scanner.close();
+        return false;
+    }
+
+    private void dumpQuiz(Quiz quiz) {
```

Comment:
インスタンスメンバにアクセスしていないため、staticでも良さそうな気がします

```suggestion
    private static void dumpQuiz(Quiz quiz) {
```